app/pages/lab-pages-editor/DataManager.jsx

```diff
@@ -0,0 +1,97 @@
+/*
+Data Manager
+Responsible for fetching, updating, and saving Workflow resources from/to
+the Panoptes service.
+ */
+
+import React, { useEffect, useMemo, useState } from 'react';
+import PropTypes from 'prop-types';
+import apiClient from 'panoptes-client/lib/api-client';
+import { WorkflowContext } from './context.jsx';
+
+function DataManager({
+  // key: to ensure DataManager renders FRESH (with states reset) whenever workflowId changes, use <DataManager key={workflowId} ... />
+  children,
+  workflowId
+}) {
+  let initialised = false;
+  const [apiData, setApiData] = useState({
+    workflow: null,
+    status: 'ready'
+  });
+
+  async function fetchWorkflow() {
```

Comment:
This line declares a new `fetchWorkflow` function each time that the component renders. Be careful about that.

Reply:
Noted, but some inefficiency is acceptable for now. I'm still considering how to organise the data fetch & _update_ functions. See Dev Notes above about "aborted useApiData() hook".